subsys/caf/modules/buttons.c

```diff
@@ -102,7 +102,7 @@ static int get_rows(uint32_t *mask)
 
 static int set_trig_mode(void)
 {
-	gpio_flags_t flags = (IS_ENABLED(CONFIG_DESKTOP_BUTTONS_POLARITY_INVERSED) ?
+	gpio_flags_t flags = (IS_ENABLED(CONFIG_CAF_BUTTONS_POLARITY_INVERSED) ?
```

Comment:
It looks like this code isn't respecting the devicetree flags that specify whether the logical signal is active high or active low.  This may confuse developers who configure that level in devicetree.

Reply:
Hi @pabigot , thanks for having a look. You are right that this is essentially moving the code from location A to B. Refactoring is done to make it usable by other applications that are on the way.

The initial goal is to move it to reuse the code. Due to timelines we have I would like to avoid big changes (if making it reusable will take too long we will not be able to reuse it for the apps we work on).
However after code is made common we would like it to fit in various cases. And for that reason we will need to plan further refactoring, updating and improving. For nrf_desktop alone there would be no point to do it.

> It looks like this code isn't respecting the devicetree flags that specify whether the logical signal is active high or active low. This may confuse developers who configure that level in devicetree.
That is true. When it was written there was no way of pass the gpio flags in dt. This is one of the limitations that we will need to fix. I would prefer to fix it in the following changes. We can do it in parallel to app development.

In the nrf_desktop there are several modules that are potentially transferable to caf (I have selected these two as I find them simplest and also because we will find the use for them rightaway). When moving the modules we will need to identify nrf_desktop specifics and get rid of them. Secondly we will need to find things that are outdated due to zephyr moving foreward and update them. We won't be able however to fix everything at once.